### PR TITLE
Allow controller changes to be received by plugin UI

### PR DESCRIPTION
**Pull request by jpcima**
This is the whole mechanism and API to receive controller change.
The value -1 for ccNumber indicates a reset of all controllers.
I will add some tests.

### CODE REVIEW

**Comment by jpcima** [comment]
Rebase to develop with fixes so that AU can manage to build again.
It's good.

File: src/sfizz/MidiState.h [hunk 1]
```diff
@@ -133,9 +134,51 @@ class MidiState
...
+        virtual ~ControllerChangeObserver() {}
+        virtual void onAllControllersReset() noexcept = 0;
+        virtual void onControllerChange(int ccNumber, float ccValue) noexcept = 0;
+    };
+
+    /**
+     * @brief Observer of controller change events, which records the
+     * information in a linked list of non-duplicated controller changes.
+     */
+    class ControllerChangeRecorder : public ControllerChangeObserver {
```
Comment (paulfd) [inline src/sfizz/MidiState.h:159]: Should this class be within the MidiState one?

Comment (jpcima) [reply src/sfizz/MidiState.h:159]: How you prefer.. I certainly don't have issue with the way how it is.

**Review by paulfd — APPROVED** [review]

**Comment by jpcima** [comment]
will be made obsolete by OSC

**Pull request closed without merging**